### DISCUSSION TITLE
[Feature] Request: Direct Album Addition and Selective Upload

### DISCUSSION THREAD

**Discussion by WoutAndAbout** [immich-app/immich#11101] [Feature Request]
### I have searched the existing feature requests to make sure this is not a duplicate request.

- [X] Yes

### The feature

Description:

Currently, to add photos to an album on the Immich mobile app, you must upload them first, which is cumbersome. It would be better if users could add photos directly to an album without this step. Additionally, for those who don't want to back up all their photos but still want to complete their albums, there should be an option to upload photos directly from their phone library to an album.

Proposed Enhancements:

1. Direct Album Addition:
   - Allow users to select and add photos to an album without needing to upload them first. They will get priority on other pictures to be uploaded first if it is a shared album so the other users can see the pictures as fast as possible.

2. Selective Upload Option:
   - Provide an option to upload photos directly from the phone library to an album without backing up all photos.

Benefits:

- Ease of Use: Makes adding photos to albums quicker and simpler.
- Convenience for Non-Backup Users: Easier for users who don’t use Immich for backups but still want to organize photos in albums on the app.

If there is anything unclear, just let me know so i can try to explain it better.

### Platform

- [ ] Server
- [ ] Web
- [X] Mobile

**Comment by bo0tzz** [comment]
#1678

**Reply by germancasares** [reply]
Not exactly the same though

**Reply by thebashsupremacy** [reply]
This is not a duplicate of the other ticket. It shouldn't be closed. It's definitely a feature still missing from the mobile app.

**Comment by WoutAndAbout** [comment]
FR [#1678 ](https://github.com/immich-app/immich/discussions/1678) is not the same.
This feature will be able to add specific pictures from your phone liberary directly into an album instead of "coping" your phone album to an immich album.

This will make it easyer to non immich users (people who don't backup all there pictures to immich) to directly upload pictures to an album without doing it thru the webinterface where it can easly fail when for instance your internet connection fails.

Lmk if smth isn't clear.

Btw thanks for making Immich sutch a great product, i love it so mutch

**Comment by echondropoulos** [comment]
This is a must have feature for myself. I share a private Immich instance with about a dozen family members and many refuse to use the app due to not being able to upload photos directly, or they simply do not understand how to upload a custom folder/album without uploading their entire phone.

Frankly I am surprised this isn't already available within the app.

**Comment by fett327** [comment]
Over a year after this feature was requested and I would also like to endorse it.  

This is exactly the way I used Google Photos before I recently switched to Immich.  I would not upload everything that my phone took.  Instead I would curate photos of kids/family and upload them into individual albums.  Immich currently supports this, but I must first selectively upload the photos to my server, and THEN reselect specific photos for an album.
Its pretty cumbersome and I dont know if my wife will make the transition so easily (which would suck since she also takes many photos of the kids).

Hoping this is in Immich's future?  Or perhaps there is a different use case I dont know about?  Regardless, thanks again to the devel team for the continued support of this awesome product.

**Comment by Xornop** [comment]
I'm frustrated that this feature is being requested many times and each time the devs just close the topic instead of listening to the community. Funny, I wanted to move away from other services because they didnt listen to their community.

**Comment by timonrieger** [comment]
dupe of #12190, please upvote instead of commenting there. Thanks!
